pyccel/codegen/printing/ccode.py

```diff
@@ -551,6 +553,11 @@ def _print_PythonBool(self, expr):
         return '({} != 0)'.format(value)
 
     def _print_Literal(self, expr):
+        if isinstance(expr, LiteralInteger) and get_final_precision(expr) == 8:
+            self.add_import(c_imports['stdint'])
+            return f"INT64_C({repr(expr.python_value)})"
+        elif isinstance(expr, LiteralFloat) and get_final_precision(expr) == 4:
+            return f"{repr(expr.python_value)}f"
         return repr(expr.python_value)
```

Comment:
I think it would be clearer and safer to break this into `_print_LiteralInteger` and `_print_LiteralFloat`